### PR TITLE
Refactoring new dims and coords, and deprecation warning for Python 2

### PR DESCRIPTION
**Pull request by roxyboy**

### CODE REVIEW

**Comment by rabernat** [comment]
I was thinking we could do something like this:

```python
def freq_for_dim(Nx, dx, real, shift):
    # figure out frequency
    return k
```


```

**Comment by roxyboy** [comment]
@lanougue Do you think you can incorporate `shift_phase` in the `_freq` function?

File: xrft/xrft.py [hunk 1]
```diff
@@ -202,9 +203,52 @@ def _transpose(da, real, trans=False):
...
+        # consistent with np.fft.rfftn
+        fftfreq = [np.fft.fftfreq]*(len(N)-1)
+        fftfreq.append(np.fft.rfftfreq)
+
+    k = [fftfreq(Nx, dx) for (fftfreq, Nx, dx) in zip(fftfreq, N, delta_x)]
+
+    if shift:
+        k = [np.fft.fftshift(l) for l in k]
+
+    return k
```
Comment (rabernat) [inline xrft/xrft.py]: Ok so now we have the frequency logic all by itself. @lanougue - do you think there is something here that needs to be fixed?

Comment (lanougue) [reply xrft/xrft.py]: It seems ok. We can leave the shifting of thecoords  outside of this function (for ifft)

Comment (lanougue) [reply xrft/xrft.py]: When defining newdims names in  _new_dims_and_coords() function, do you want to keep something as 'freq_freq_x' as a possibility ?
I my PR, I proposed something as
'''
newdims[anum] = prefix + d if d[:len(prefix)]!=prefix else d[len(prefix):]
'''
which will transform back 'freq_x' into 'x'

Comment (roxyboy) [reply xrft/xrft.py]: I agree that we should probably avoid the possibility such as `freq_freq_x`. The `prefix` argument in `dft` should make this flexible?

Comment (roxyboy) [reply xrft/xrft.py]: I adopted your code: `newdims[anum] = prefix + d if d[:len(prefix)]!=prefix else d[len(prefix):]`

Comment (roxyboy) [reply xrft/xrft.py]: @lanougue Are you suggesting that we incorporate the shifting of the coords for ifft for the next round of PR after merging this one? I think we should incorporate your enhancements!

Comment (lanougue) [reply xrft/xrft.py]: No, what I mean is that shifting the coords in ifft should be done outside the _freq() function.
It definitively pertains to ifft() function

**Comment by rabernat** [comment]
Ok, LGTM!

**Comment by lanougue** [comment]
Reopening this point.

If original coordinate name is something like "freq_x", applying xrft.dft return a "x" coordinate, which is the expected behaviour.
However, the spacing attribute is not correctly named. Instead of "x_spacing" we get "freq_freq_x_spacing.

Since @roxyboy was in charge of this modification, I let him correct this.
cheers

**Comment by lanougue** [comment]
I did a new PR Phase corrections #122  to solve other things with the forward transform.
I finally also added corrections mentioned above about naming new coordinates and spacing in _new_dims_and_coords() function.
@rabernat , @roxyboy , if you can take a look

**Pull request merged**